### ISSUE TITLE
Use case missing for unused-argument check

### ISSUE THREAD

**Issue by sushobhit27** [PyCQA/pylint#2146]
### Steps to reproduce
1. $ cat sample1.py
```python
def foo(a, b):
    a = 666
    return a + b
```
2. $ pylint sample1.py --disable=blacklisted-name --enable=unused-argument

### Current behavior
-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 6.67/10, +3.33)

### Expected behavior
There should be a warning that original value from parameter 'a' is not used or overridden.

### pylint --version output
pylint 2.0.0
astroid 1.6.3
Python 3.6.4 (default, Jan  7 2018, 15:53:53)
[GCC 6.4.0]

**Comment by PCManticore** [comment]
Makes sense to have this as a check or as an addition to `unused-argument` as it might result in a surprising bug if the function is complicated enough.

**Comment by sushobhit27** [comment]
@PCManticore ok I will pick up this issue then.

**Comment by sushobhit27** [comment]
@PCManticore Is there something like is_name_used_in_parent_scope in utils, as this functionality will help in resolving both this and #2223

**Comment by clavedeluna** [comment]
I believe this is a duplicate of #3102

**Comment by Pierre-Sassoulas** [comment]
Closing as duplicate because the discussion in the other issue is more detailed, thank you @clavedeluna .
